app/Http/Controllers/Auth/SocialiteController.php

```diff
@@ -52,6 +52,11 @@ public function redirect(Request $request, string $provider)
         // Re-store target url since it will be forgotten after the redirect
         $request->session()->put('url.intended', redirect()->intended()->getTargetUrl());
 
+        if( $provider == "okta" && LibreNMSConfig::get('auth.socialite.configs.okta.group_claim', false) ){
```

Comment:
This feels out of place.
It is complaining below because the interface does not support "scopes".  So either check that it is the Okta driver, or that "scopes" exists.

I'm not sure if there is a generic way of doing this or not.

Reply:
I don't believe there is a generic way, but I could be wrong.  checking 
```$driver instanceof \Laravel\Socialite\Two\AbstractProvider```
should mean that ->scopes() is valid ... but it could mean different things for different providers (probably) 

I would hope that if these are available for example from Git or Microsoft etc that the next person will deal with appropriately, and should some generic or repeated patterns appear then adjust as appropriate?

Reply:
I looked through providers the other day and there were very few with group support and the documentation of groups sucked.

Reply:
I looked at docs for GitHub and Azure AD as well.    It appears with Azure AD you can define 'roles' on the application and have these sent if the claim is 'role', and also a wait to have AD groups (potentially filtered) via 'groups'.

Since I don't have a Azure AD lying around I can't actually see what would be returned, so it might not work 'out-of-the-box' but it in theory shouldn't be too hard if the data is somewhere in the user structure.